common/src/main/java/tachyon/ClientBase.java

```diff
@@ -189,15 +189,22 @@ public synchronized void disconnect() {
       mConnected = false;
     }
     try {
-      if (mProtocol != null) {
-        mProtocol.getTransport().close();
-      }
+      closeOperation();
     } finally {
       afterDisconnect();
     }
   }
 
   /**
+   * Close the connection.
+   */
+  protected void closeOperation() {
```

Comment:
How about still have
```java
if (mProtocol != null) {
   mProtocol.getTransport().close();
}```
inside `disconnect`, but before that piece of code, calling a method named `beforeDisconnect`.  `beforeDisconnect` is an empty method in `ClientBase`, but meant to be a hook method for subclasses, just like `afterConnect`. I feel this approach might be a bit simpler and easier to understand.